File: c++/leetcode/1380. Lucky Numbers in a Matrix.cpp
class Solution
{
public:
  vector<int> luckyNumbers(vector<vector<int>> &matrix)
  {
    vector<int> ans;

    int k = matrix.size();
    int col = matrix[0].size();
    int row = 0;

    while (row < k)
    {
      auto setcol = 0;
      auto _min = matrix[row][0];

      for (int j = 0; j < col; ++j)
      {
        if (matrix[row][j] < _min)
        {
          _min = matrix[row][j];
          setcol = j;
        }
      }

      auto _max = matrix[0][setcol];
      for (int i = 0; i < k; ++i)
        if (matrix[i][setcol] > _max)
          _max = matrix[i][setcol];

      if (_max == _min)
        ans.push_back(_max);

      ++row;
    }

    return ans;
  }
};

File: javascript/leetcode/1380. Lucky Numbers in a Matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
var luckyNumbers = function (matrix) {
  const ans = [];

  let k = matrix.length;
  let col = matrix[0].length;
  let row = 0;

  while (row < k) {
    let setcol = 0;
    let _min = matrix[row][0];

    for (let j = 0; j < col; ++j) {
      if (matrix[row][j] < _min) {
        _min = matrix[row][j];
        setcol = j;
      }
    }

    let _max = matrix[0][setcol];
    for (let i = 0; i < k; ++i) {
      if (matrix[i][setcol] > _max) _max = matrix[i][setcol];
    }

    if (_max === _min) ans.push(_max);

    ++row;
  }

  return ans;
};
